disnake/shard.py

```diff
@@ -330,12 +333,41 @@ class AutoShardedClient(Client):
     if TYPE_CHECKING:
         _connection: AutoShardedConnectionState
 
+    @overload
     def __init__(
-        self, *args: Any, loop: Optional[asyncio.AbstractEventLoop] = None, **kwargs: Any
-    ) -> None:
-        kwargs.pop("shard_id", None)
-        self.shard_ids: Optional[List[int]] = kwargs.pop("shard_ids", None)
-        super().__init__(*args, loop=loop, **kwargs)
+        self,
+        *,
+        asyncio_debug: bool = False,
+        loop: Optional[asyncio.AbstractEventLoop] = None,
+        shard_ids: List[int] = None,  # instead of Client's shard_id: Optional[int]
```

Comment:
```suggestion
        shard_ids: Optional[List[int]] = None,  # instead of Client's shard_id: Optional[int]
```

Reply:
Fixed in 83ddb4a3785dc78c0d97c8c33f77c5c6dbcff996.